training/train.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+
+import torch
+import torchvision
+import torchvision.transforms.v2 as transforms
+from torch.utils import data
+
+from data.dataset import Dataset
+from models.centernet import ModelBuilder, input_height, input_width
+from training.encoder import CenternetEncoder
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-o", "--overfit", action="store_true", help="overfit to 10 images")
+args = parser.parse_args()
+
+overfit = args.overfit
+
+dataset_val = torchvision.datasets.VOCDetection(
+    root="../VOC", year="2007", image_set="val", download=False
+)
+
+transform = transforms.Compose(
+    [
+        transforms.Resize(size=(input_width, input_height)),
+        transforms.ToImage(),
+        transforms.ToDtype(torch.float32, scale=True),
+    ]
+)
+
+encoder = CenternetEncoder(input_height, input_width)
+
+dataset_val = torchvision.datasets.wrap_dataset_for_transforms_v2(dataset_val)
+torch_dataset = Dataset(dataset=dataset_val, transformation=transform, encoder=encoder)
+
+training_data = torch_dataset
+lr = 0.001
+desired_loss = 0.1
```

Comment:
For train mode it is impossible value. So if overfit =False, then train will be infinite. It is better to set learning rate schedule for overfit and for train modes

Reply:
Done

Reply:
For train mode we have desired_loss = 0.1 but it is an impossible requirement. Please, set max_epoch as the criterion to finish train mode

Reply:
Fixed